File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] res = new int[2];
        
        for(int i = 0; i < n; i++){
            if(map.containsKey(target - nums[i])){
                res[0] = i;
                res[1] = map.get(target - nums[i]);
                
                break;
            }
            
            map.put(nums[i], i);
            
            
        }
        
        
        return res;
    }
}

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    var m map[int]int

    // create a slice of array with 2 elements
    res := make([]int, 2)

    m = make(map[int]int)

    for i, num := range nums {
        x, ok := m[target - num];

        if ok {
            res[0] = i;
            res[1] = x;

            break;
        }

        m[num] = i;


    }

    return res;
    
}